core/src/main/java/sunstone/core/SunstoneConfig.java

```diff
@@ -0,0 +1,155 @@
+        }
+        return (V) defaultValue.getClass().cast(value);
+    }
+
+    /**
+     * Resolve expression. Resolve ${.*} into other config properties.
+     *
+     * {@code clazz} is used as converter in {@link SmallRyeConfig#getValue(String, Class)} which is ultimately used for
+     * resolving expression.
+     */
+    public static <V> V resolveExpression(String expression, Class<V> clazz) {
+        return config.getValue(SunstoneExpressionSource.registerExpression(expression), clazz);
+    }
+
+
+    /**
+     * Resolve expression to String. If expression is blank, it is returned as is.
+     * Resolve ${.*} into other config properties.
+     */
+    public static String resolveExpressionToString(String expression) {
+        return expression.isBlank() ? expression : resolveExpression(expression, String.class);
+    }
+
+    /**
+     * Returns true if string contains ${.*}
+     */
+    public static boolean isExpression(String candidate) {
+        return candidate.matches("^.*\\$\\{.*\\}.*$");
+    }
+
+    /**
+     * Source for sunstone.properties.
+     */
+    static class SunstonePropertiesSource implements ConfigSource {
+        public static final String DEFAULT_PROPERTIES = "/sunstone.properties";
+        public Properties load() {
+            Properties properties = new Properties();
+            // reload default properties
+            if (SunstonePropertiesSource.class.getResource(SunstonePropertiesSource.DEFAULT_PROPERTIES) != null) {
+                try (InputStream is = SunstonePropertiesSource.class.getResourceAsStream(SunstonePropertiesSource.DEFAULT_PROPERTIES)) {
+                    if (is != null) {
+                        properties.load(is);
+                    }
+                } catch (IOException e) {
+                    SunstoneLogger.DEFAULT.error("Unable to load default properties", e);
+                }
+            }
+            return properties;
+        }
+
+        @Override
+        public Set<String> getPropertyNames() {
+            return load().stringPropertyNames();
+        }
+
+        @Override
+        public String getValue(String propertyName) {
+            return load().getProperty(propertyName);
+        }
+
+        @Override
+        public String getName() {
+            return "sunstone-source";
+        }
+    }
+
+    /**
+     * Fake SR Config source for resolving expressions.
+     * To resolve expression, you need to:
+     *  - Register {@link SunstoneExpressionSource} as a source for SR Config
+     *  - SR Config must be created with correct interceptors so that expressions are resolver
+     *  - register expression as a new property in internal map and get key to retrieve it - {@link #registerExpression(String)}
+     *  - retrieve the value ({@link SmallRyeConfig#getValue(String, Class)}) using key provided by {@link #registerExpression(String)}
+     */
+    static class SunstoneExpressionSource implements ConfigSource {
```

Comment:
I would need to explain this. Why is expression special case? It is related to another thread. Why we need special API for dealing with expressions?

Reply:
I wan't to have `@AzureVirtualMachine(name="jboss-${test.run.name}")`. I need it to be resolved. With this I can make SR Config to resolve it.

Reply:
This can be marked as Resolved. We discuss that in other thread.